vision/odlc_contour_filtering.py

```diff
@@ -0,0 +1,407 @@
+
+    returns
+    -------
+    returns true if the aspect ratio of the min area box is inbetween the min and max
+    """
+    min_area_box: min_area_box_type = cv2.boxPoints(cv2.minAreaRect(contour))
+    aspect_ratio: float = (cv2.norm(min_area_box[0] - min_area_box[1])) / (
+        cv2.norm(min_area_box[1] - min_area_box[2])
+    )
+
+    return min_box_ratio < aspect_ratio < max_box_ratio
+
+
+def test_bounding_box(
+    contour: contour_type, dims: tuple[int, int], test_area_ratio: float = 10.0
+) -> bool:
+    """
+    calculates the area of a upright bounding box around the given contour
+    and compares it to the rectangle (image) of the given dimensions
+
+    parameters
+    ----------
+    contour : contour_type
+        The individual contour to be evaluated (as returned from cv2.findContours)
+    dims : tuple[int, int]
+        The dimensions of the image the contour is from
+    test_area_ratio : float
+        The minimum acceptable ratio of image area to contour bounding box area
+
+    returns
+    -------
+    returns true if the bounding box area is less than the image area by a factor of
+    test_area_ratio or more
+    """
+    bounding_box: bound_box_type = cv2.boundingRect(contour)
+
+    box_area: float = abs(bounding_box[2] - bounding_box[0]) * abs(
+        bounding_box[3] - bounding_box[1]
+    )
+    img_area: float = dims[0] * dims[1]
+
+    return box_area * test_area_ratio <= img_area
+
+
+def test_spikiness(contour: contour_type) -> bool:
+    """
+    Checks whether the contour has some points that are much farther away from the center of mass
+    than the rest of the points
+    basically real shapes will not have random points that are really far away from the rest
+    but a weird patch of grass or something without well defined edges will have crazy points
+    all over the place
+
+    parameters
+    ----------
+    contour : contour_type
+        The individual contour to be evaluated (as returned from cv2.findContours)
+
+    returns
+    -------
+    True unless there is a statistical outlier point that is uniquely far away (or close ig)
+    """
+    moments: dict[str, float] = cv2.moments(contour)
+    # com is Center of Mass, com = (y_coord, x_coord)
+    com: NDArray[Shape["2"], Float32] = np.array(
+        ((moments["m01"] / moments["m00"]), (moments["m10"] / moments["m00"])), dtype=np.float64
+    )
+
+    dists_com: NDArray[Shape["*"], Float32] = np.ndarray(contour.shape[0], Float32)
+    for i in range(contour.shape[0]):
+        dists_com[i] = cv2.norm(contour[i] - com)
+
+    dists_median: float = np.median(dists_com)
+    dists_outlier_range: float = 1.5 * (np.quantile(dists_com, 0.75) - np.quantile(dists_com, 0.25))
+
+    if np.all(dists_com < dists_median + dists_outlier_range):
+        return True
+    return False
+
+
+def _min_common_bounding_box(contours: tuple[contour_type, ...]) -> bound_box_type:
+    """
```

Comment:
It looks like the only place you are calling this function is a case where the tuple always has 2 elements. If that will always be the case, then the type should specify that instead of ellipsis. If the number of elements can vary, consider using a list instead.